src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_SIZE = 16;
+    private static final int RESIZE = 2;
+    private static final double THRESHOLD = 0.75;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_SIZE];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkForResize();
+        Node<K, V> newNode = new Node<>(key, value);
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
```

Comment:
loks like you can merge this lines like "Node<K, V> node = table[getIndex(key)];"